Src/RhinoPlugin.Tests/TestInit.cs

```diff
@@ -1,82 +1,56 @@
-        static string systemDir = null;
-        static string systemDirOld = null;
+        static string rhinoDir;
 
         [AssemblyInitialize]
         public static void AssemblyInitialize(TestContext context)
         {
+            //get the correct rhino 7 installation directory
+            rhinoDir = Registry.GetValue(@"HKEY_LOCAL_MACHINE\SOFTWARE\McNeel\Rhinoceros\7.0\Install", "Path", null) as string ?? string.Empty;
+            Assert.IsTrue(System.IO.Directory.Exists(rhinoDir), "Rhino system dir not found: {0}", rhinoDir);
+            context.WriteLine(" The current Rhino 7 installation is " + rhinoDir);
+
             if (initialized)
             {
-                throw new InvalidOperationException("AssemblyInitialize should only be called once");
+                throw new InvalidOperationException("Initialize Rhino.Inside once");
             }
-            initialized = true;
-
-            context.WriteLine("Assembly init started");
-
-            // Ensure we are 64 bit
+            else
+            {
+                RhinoInside.Resolver.Initialize();
+                initialized = true;
+                context.WriteLine("Rhino.Inside init has started");
+            }
+            
+            // Ensure we are running the tests in x64
             Assert.IsTrue(Environment.Is64BitProcess, "Tests must be run as x64");
 
             // Set path to rhino system directory
             string envPath = Environment.GetEnvironmentVariable("path");
-            string programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
-            systemDir = System.IO.Path.Combine(programFiles, "Rhino 7 WIP", "System");
-            systemDirOld = System.IO.Path.Combine(programFiles, "Rhino WIP", "System");
-            if (System.IO.Directory.Exists(systemDir) != true)
-            {
-                systemDir = systemDirOld;
-            }
-
-
-            Assert.IsTrue(System.IO.Directory.Exists(systemDir), "Rhino system dir not found: {0}", systemDir);
-
-            // Add rhino system directory to path (for RhinoLibrary.dll)
-            Environment.SetEnvironmentVariable("path", envPath + ";" + systemDir);
+            Environment.SetEnvironmentVariable("path", envPath + ";" + rhinoDir);
 
-            // Add hook for .Net assmbly resolve (for RhinoCommmon.dll)
-            AppDomain.CurrentDomain.AssemblyResolve += ResolveRhinoCommon;
-
-            // Start headless Rhino process
-            LaunchInProcess(0, 0);
-        }
-
-        private static Assembly ResolveRhinoCommon(object sender, ResolveEventArgs args)
-        {
-            var name = args.Name;
-
-            if (!name.StartsWith("RhinoCommon"))
-            {
-                return null;
-            }
-
-            var path = System.IO.Path.Combine(systemDir, "RhinoCommon.dll");
-            return Assembly.LoadFrom(path);
+            // Start a headless rhino instance using Rhino.Inside
+            StartRhino();
```

Comment:
suggest dropping `StartRhino` and inlining this call instead. If you need to save the result to a variable to prevent it being garbage collected then make `_rhinoCore` into a static var.
```suggestion

_rhinoCore = new Rhino.Runtime.InProcess.RhinoCore(null, Rhino.Runtime.InProcess.WindowStyle.NoWindow);
```

Reply:
@tmakin I had this inclined initially. However it seems that for rhino to be loaded properly we absolutely need the `STAThreadAttribute` which can only decorate methods.

- tidied up the code 
- cleaned up the extra summary elements
- upgraded to the new csproj format 
- this somehow made gh not to load properly so I added the GH assembly resolution for the the VS test class as well

Reply:
Excellent, thanks. I've just tried the latest version and all the MsTest tests are passing now 🎉